google-cloud-bigquery/lib/google/cloud/bigquery/table.rb

```diff
@@ -512,6 +519,53 @@ def clustering_fields
           @gapi.clustering.fields if clustering?
         end
 
+        ##
+        # Updates the list of fields on which data should be clustered.
+        #
+        # Only top-level, non-repeated, simple-type fields are supported. When
+        # you cluster a table using multiple columns, the order of columns you
+        # specify is important. The order of the specified columns determines
+        # the sort order of the data.
+        #
+        # BigQuery supports clustering for both partitioned and non-partitioned
+        # tables.
+        #
+        # See {Table::Updater#clustering_fields=}, {Table#clustering_fields} and
+        # {Table#clustering?}.
+        #
+        # @see https://cloud.google.com/bigquery/docs/clustered-tables
+        #   Introduction to clustered tables
+        # @see https://cloud.google.com/bigquery/docs/creating-clustered-tables
+        #   Creating and using clustered tables
+        # @see https://cloud.google.com/bigquery/docs/creating-clustered-tables#modifying-cluster-spec
+        #   Modifying clustering specification
+        #
+        # @param [Array<String>, nil] fields The clustering fields, or `nil` to
```

Comment:
Would the empty array be meaningful here? (Should it have the same effect as `nil`?)

Reply:
Empty array currently gets a "domain-specific" server-side error. We don't usually document such errors since they might change.